tests/foreman/ui_airgun/test_contentview.py

```diff
@@ -1571,6 +1569,210 @@ def test_positive_remove_cv_version_from_default_env(session, module_org):
+            )
+        ]
+    )
+    repo.setup(module_org.id)
+    cv, lce = repo.setup_content_view(module_org.id)
+    with session:
+        cv_values = session.contentview.read(cv['name'])
+        assert cv_values['puppet_modules']['table'][0]['Name'] == puppet_module_name
+        assert cv_values['puppet_modules']['table'][0]['Author'] == author
+        cvv = session.contentview.read_version(cv['name'], VERSION)
+        assert puppet_module_name in cvv['puppet_modules']['table'][0]['Name']
+        assert cvv['puppet_modules']['table'][0]['Author'] == author
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert ENVIRONMENT in cvv['Environments']
+        # remove the content view version from Library
+        session.contentview.remove_version(cv['name'], VERSION, False, [ENVIRONMENT])
+        cvv = session.contentview.search_version(cv['name'], VERSION)[0]
+        assert ENVIRONMENT not in cvv['Environments']
+        # ensure that puppet module still in content view version
+        cvv = session.contentview.read_version(cv['name'], VERSION)
+        assert puppet_module_name in cvv['puppet_modules']['table'][0]['Name']
+        assert cvv['puppet_modules']['table'][0]['Author'] == author
+
+
+@tier2
+def test_positive_remove_qe_promoted_cv_version_from_default_env(session, module_org):
+    """Remove QE promoted content view version from Library environment
+
+    :id: 71ad8b72-68c4-4c98-9387-077f54ef0184
+
+    :Steps:
+
+        1. Create a content view
+        2. Add docker repo(s) to it
+        3. Publish content view
+        4. Promote the content view version to multiple environments
+            Library -> DEV -> QE
+        5. remove the content view version from Library environment
+
+    :expectedresults: Content view version exist only in DEV, QE and not in
+        Library
+
+    :CaseLevel: Integration
+    """
+    dev_lce = entities.LifecycleEnvironment(organization=module_org).create()
+    qe_lce = entities.LifecycleEnvironment(organization=module_org, prior=dev_lce).create()
+    repo = RepositoryCollection(
+        repositories=[DockerRepository(
+            url=DOCKER_REGISTRY_HUB, upstream_name=DOCKER_UPSTREAM_NAME)
+        ]
+    )
+    repo.setup(module_org.id)
+    repo_name = repo.repos_info[0]['name']
+    cv, lce = repo.setup_content_view(module_org.id, dev_lce.id)
+    cvv = entities.ContentView(id=cv['id']).read().version[0]
+    promote(cvv, qe_lce.id)
+    with session:
+        cv_values = session.contentview.read(cv['name'])
+        assert cv_values['docker_repositories']['resources']['assigned'][0]['Name'] == repo_name
+        cvv_content = session.contentview.read_version(cv['name'], VERSION)
+        assert cvv_content['docker_repositories']['table'][0]['Name'] == repo_name
+        cvv_table = session.contentview.search_version(cv['name'], VERSION)
+        assert ' '.join((ENVIRONMENT, dev_lce.name, qe_lce.name)) == cvv_table[0]['Environments']
```

Comment:
Are you are sure, they will be always in the same order?

Reply:
yes. If no - then it is a bug and we should look on that